seacrowd/sea_datasets/wongnai_reviews/wongnai_reviews.py

```diff
@@ -0,0 +1,114 @@
+import os
+import csv
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+# no BibTeX citation
+_CITATION = ""
+
+_DATASETNAME = "wongnai_reviews"
+
+_DESCRIPTION = """
+Wongnai features over 200,000 restaurants, beauty salons, and spas across Thailand on its platform, with detailed
+information about each merchant and user reviews. Its over two million registered users can search for what’s top rated
+in Bangkok, follow their friends, upload photos, and do quick write-ups about the places they visit. Each write-up
+(review) also comes with a rating score ranging from 1-5 stars. The task here is to create a rating prediction model
+using only textual information.
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/wongnai_reviews"
+
+_LANGUAGES = ["tha"]
+
+_LICENSE = Licenses.LGPL_3_0.value
+
+_LOCAL = False
+
+_URLS = {_DATASETNAME: "https://archive.org/download/wongnai_reviews/wongnai_reviews_withtest.zip"}
+
+_SUPPORTED_TASKS = [Tasks.ASPECT_BASED_SENTIMENT_ANALYSIS]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class WongnaiReviewsDataset(datasets.GeneratorBasedBuilder):
+    """WongnaiReviews consists reviews for over 200,000 restaurants, beauty salons, and spas across Thailand."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_text_multi",
```

Comment:
following the task type, the schema (and therefore the config name) should be `seacrowd_text`. This also applies here and throughout the code.